builder/frameworks/espidf.py

```
@@ -195,177 +213,205 @@ def _get_installed_standard_pip_packages():
+    def write_sdkconfig_file(idf_config_flags, checksum_source):
+        if "arduino" not in env.subst("$PIOFRAMEWORK"):
+            print("Error: Arduino framework required for sdkconfig processing")
+            return
+        """Write the final sdkconfig.defaults file with checksum."""
+        sdkconfig_src = join(arduino_libs_mcu, "sdkconfig")
+        sdkconfig_dst = join(PROJECT_DIR, "sdkconfig.defaults")
+        
+        # Generate checksum for validation (maintains original logic)
+        checksum = get_MD5_hash(checksum_source.strip() + mcu)
+        
+        with open(sdkconfig_src, 'r') as src, open(sdkconfig_dst, 'w') as dst:
```

Pylint (3.3.7)

[warning] Using open without explicitly specifying an encoding

(W1514)